streams_map.go

```diff
@@ -164,17 +165,19 @@ func (m *streamsMap) OpenStream() (*stream, error) {
 
 // AcceptStream returns the next stream opened by the peer
 // it blocks until a new stream is opened
-// TODO: implement error conditions
 func (m *streamsMap) AcceptStream() (utils.Stream, error) {
 	m.mutex.Lock()
 	var str utils.Stream
 	for {
 		var ok bool
+		if m.closeErr != nil {
+			return nil, m.closeErr
```

Comment:
The mutex is not unlocked in this return path. Insert a `defer m.mutex.Unlock()` right after locking and remove the `Unlock()` below.